src/main/java/org/folio/fqm/service/FqlToSqlConverterService.java

```diff
@@ -135,7 +141,17 @@ private static Condition handleEquals(EqualsCondition equalsCondition, EntityTyp
       return handleDate(equalsCondition, field);
     }
     String dataType = getFieldDataType(entityType, equalsCondition);
-    if (STRING_TYPE.equals(dataType) || RANGED_UUID_TYPE.equals(dataType) || OPEN_UUID_TYPE.equals(dataType) || DATE_TYPE.equals(dataType))  {
+    String filterFieldDataType = getFieldForFiltering(equalsCondition, entityType).getDataType().getDataType();
+    if (RANGED_UUID_TYPE.equals(filterFieldDataType) || OPEN_UUID_TYPE.equals(filterFieldDataType)) {
+      String value = (String) equalsCondition.value();
+      if (!isValidUUID(value)) {
+        Map<String, String> errors = Map.of(INVALID_UUID, ERROR_MESSAGE + value);
+        throw new InvalidFqlException(field + equalsCondition.toString(), errors);
+      }
+      return cast(field, UUID.class).eq(cast(value, UUID.class));
+    }
+
+    if (STRING_TYPE.equals(dataType) || RANGED_UUID_TYPE.equals(dataType) || OPEN_UUID_TYPE.equals(dataType) || DATE_TYPE.equals(dataType) || STRING_UUID_TYPE.equals(dataType))  {
```

Comment:
We don't need the RANGED_UUID_TYPE or OPEN_UUID_TYPE conditions here anymore, do we? Since they'll be handled above.